pkg/network/tracer/tracer_usm_linux_test.go

```diff
@@ -1654,44 +1717,58 @@ func testAMQPProtocolClassification(t *testing.T, tr *Tracer, clientHost, target
 		require.NoError(t, client.DeleteQueues())
 	}
 
+	var preTracerSetupWrapper func(func(*testing.T, testContext)) func(*testing.T, testContext)
+	if withTLS {
+		// Start the proxy server.
+		authority := net.JoinHostPort(serverHost, amqpsPort)
+		cmd, cancel := proxy.NewExternalUnixTransparentProxyServer(t, unixProxyPath, authority, amqp.TLS)
+		t.Cleanup(cancel)
+		require.NoError(t, proxy.WaitForConnectionReady(unixProxyPath))
+
+		pid := cmd.Process.Pid
+		utils.WaitForProgramsToBeTraced(t, "go-tls", pid)
+
+		preTracerSetupWrapper = func(f func(t *testing.T, ctx testContext)) func(*testing.T, testContext) {
+			return func(t *testing.T, ctx testContext) {
+				goTLSDetachPID(t, pid)
+				f(t, ctx)
+				goTLSAttachPID(t, pid)
+			}
+		}
+	}
```

Comment:
1. No need to rerun the proxy server every time, you all need to attach or detach in each test
2. Why keeping a large difference between plaintext and TLS tests? you can use the proxy for all scenarios, and just enable/disable TLS

Reply:
we need to detach after the postTracer